metadata-io/src/main/java/com/linkedin/metadata/system_info/collectors/PropertiesCollector.java

```diff
@@ -0,0 +1,178 @@
+    Map<String, PropertyInfo> sortedProperties = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);
+
+    if (springEnvironment instanceof ConfigurableEnvironment) {
+      ConfigurableEnvironment configEnv = (ConfigurableEnvironment) springEnvironment;
+
+      for (PropertySource<?> propertySource : configEnv.getPropertySources()) {
+        if (propertySource instanceof EnumerablePropertySource) {
+          EnumerablePropertySource<?> enumerable = (EnumerablePropertySource<?>) propertySource;
+
+          for (String key : enumerable.getPropertyNames()) {
+            // TreeMap will keep the first occurrence (highest priority source)
+            sortedProperties.computeIfAbsent(
+                key,
+                k -> {
+                  Object rawValue = enumerable.getProperty(k);
+                  String resolvedValue = springEnvironment.getProperty(k);
+
+                  // Check if this is an allowed property
+                  if (isAllowedProperty(k)) {
+                    return PropertyInfo.builder()
+                        .key(k)
+                        .value(rawValue)
+                        .source(propertySource.getName())
+                        .sourceType(propertySource.getClass().getSimpleName())
+                        .resolvedValue(resolvedValue)
+                        .build();
+                  } else {
+                    return PropertyInfo.builder()
+                        .key(k)
+                        .value("***REDACTED***")
+                        .source(propertySource.getName())
+                        .sourceType(propertySource.getClass().getSimpleName())
+                        .resolvedValue("***REDACTED***")
+                        .build();
+                  }
+                });
+          }
+        }
+      }
+    }
+
+    return sortedProperties;
+  }
+
+  private List<PropertySourceInfo> getPropertySources() {
+    List<PropertySourceInfo> sources = new ArrayList<>();
+
+    if (springEnvironment instanceof ConfigurableEnvironment) {
+      ConfigurableEnvironment configEnv = (ConfigurableEnvironment) springEnvironment;
+
+      for (PropertySource<?> propertySource : configEnv.getPropertySources()) {
+        int propertyCount = 0;
+        if (propertySource instanceof EnumerablePropertySource) {
+          propertyCount = ((EnumerablePropertySource<?>) propertySource).getPropertyNames().length;
+        }
+
+        sources.add(
+            PropertySourceInfo.builder()
+                .name(propertySource.getName())
+                .type(propertySource.getClass().getSimpleName())
+                .propertyCount(propertyCount)
+                .build());
+      }
+    }
+
+    return sources;
+  }
+
+  private boolean isAllowedProperty(String key) {
+    String lowerKey = key.toLowerCase();
+    return ALLOWED_PATTERNS.stream().anyMatch(pattern -> pattern.matcher(lowerKey).find())
+        || SENSITIVE_PATTERNS.stream().noneMatch(lowerKey::endsWith);
```

Comment:
I copied the sensitive detection logic from the original PR, however I wonder how much we are concerned about avoiding exposing sensitive data.

I think the most safe is approach is the explicit allow list, when the element is not in the allow list it would be redacted. This allow list should not use wildcards (since with the wildcards it is easier to miss and leak sensitive data. At the same time there are almost 700 items in the config, so this explicit allowlist would be pretty large.

@david-leifker and @RyanHolstien - what are your thoughts on this?